homeassistant/components/telegram_bot/__init__.py

```diff
@@ -857,131 +858,102 @@ def leave_chat(self, chat_id=None):
-            data[ATTR_MSGID] = msg_data["message_id"]
-        if "last_name" in msg_data["from"]:
-            data[ATTR_FROM_LAST] = msg_data["from"]["last_name"]
-        if "chat" in msg_data:
-            data[ATTR_CHAT_ID] = msg_data["chat"]["id"]
-        elif ATTR_MESSAGE in msg_data and "chat" in msg_data[ATTR_MESSAGE]:
-            data[ATTR_CHAT_ID] = msg_data[ATTR_MESSAGE]["chat"]["id"]
-
-        return True, data
-
-    def _get_channel_post_data(self, msg_data):
-        """Return boolean msg_data_is_ok and dict msg_data."""
-        if not msg_data:
-            return False, None
-
-        if "sender_chat" in msg_data and "chat" in msg_data and "text" in msg_data:
-            if (
-                msg_data["sender_chat"].get("id") not in self.allowed_chat_ids
-                and msg_data["chat"].get("id") not in self.allowed_chat_ids
-            ):
-                # Neither sender_chat id nor chat id was in allowed_chat_ids,
-                # origin is not allowed.
-                _LOGGER.error("Incoming message is not allowed (%s)", msg_data)
-                return True, None
-
-            data = {
-                ATTR_MSGID: msg_data["message_id"],
-                ATTR_CHAT_ID: msg_data["chat"]["id"],
-                ATTR_TEXT: msg_data["text"],
-            }
-            return True, data
-
-        _LOGGER.error("Incoming message does not have required data (%s)", msg_data)
-        return False, None
-
-    def process_message(self, data):
-        """Check for basic message rules and fire an event if message is ok."""
-        if ATTR_MSG in data or ATTR_EDITED_MSG in data:
-            event = EVENT_TELEGRAM_COMMAND
-            if ATTR_MSG in data:
-                data = data.get(ATTR_MSG)
-            else:
-                data = data.get(ATTR_EDITED_MSG)
-            message_ok, event_data = self._get_message_data(data)
-            if event_data is None:
-                return message_ok
-
-            if ATTR_MSGID in data:
-                event_data[ATTR_MSGID] = data[ATTR_MSGID]
-
-            if "text" in data:
-                if data["text"][0] == "/":
-                    pieces = data["text"].split(" ")
-                    event_data[ATTR_COMMAND] = pieces[0]
-                    event_data[ATTR_ARGS] = pieces[1:]
-                else:
-                    event_data[ATTR_TEXT] = data["text"]
-                    event = EVENT_TELEGRAM_TEXT
-            else:
-                _LOGGER.warning("Message without text data received: %s", data)
-                event_data[ATTR_TEXT] = str(data)
-                event = EVENT_TELEGRAM_TEXT
 
-            self.hass.bus.async_fire(event, event_data)
+    def handle_update(self, update: Update, context: CallbackContext):
+        """Handle updates from bot dispatcher set up by the respective platform."""
+        _LOGGER.debug("Handling update %s", update)
+        if not self.authorize_update(update):
+            return
```

Comment:
If the return value should be a boolean we should return False here.
```suggestion
            return False
```